typed/merge.go

```diff
@@ -374,8 +392,8 @@ func (w *mergingWalker) visitMapItems(t schema.Map, lhs, rhs *value.Map) (errs V
 
 func (w *mergingWalker) doMap(t schema.Map) (errs ValidationErrors) {
 	var lhs, rhs *value.Map
-	errs = append(errs, w.derefMapOrStruct("lhs: ", "map", w.lhs, &lhs)...)
-	errs = append(errs, w.derefMapOrStruct("rhs: ", "map", w.rhs, &rhs)...)
+	w.derefMapOrStruct("lhs: ", "map", w.lhs, &lhs)
+	w.derefMapOrStruct("rhs: ", "map", w.rhs, &rhs)
```

Comment:
We could actually remove these errors from all the cases, but I wanted to keep the diff smaller

Reply:
If we don't think that they should ever happen, can we panic?

Reply:
the error case can happen, but what we want it to do in that case is treat the value as nil